File: app/src/main/java/eu/sesma/dagger/platform/ApplicationModule.java
package eu.sesma.dagger.platform;

import android.content.Context;

import eu.sesma.dagger.ui.AppCollaborator;
import eu.sesma.dagger.ui.IAppCollaborator;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    private final AndroidApplication application;

    public ApplicationModule(AndroidApplication application) {
        this.application = application;
    }

    @Provides
    Context provideContext() {
        return this.application;
    }

    @Provides
    @Singleton
    IAppCollaborator provideAppCollaborator() {
        return new AppCollaborator();
    }
}

File: app/src/main/java/eu/sesma/dagger/platform/ApplicationModule.kt
package eu.sesma.dagger.platform

import android.content.Context
import dagger.Module
import dagger.Provides
import eu.sesma.dagger.ui.AppCollaborator
import eu.sesma.dagger.ui.IAppCollaborator
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: AndroidApplication) {

    @Provides
    fun provideContext(): Context {
        return this.application
    }

    @Provides
    @Singleton
    fun provideAppCollaborator(): IAppCollaborator {
        return AppCollaborator()
    }
}
